crypt-yescrypt.c

```diff
@@ -39,32 +39,39 @@ gensalt_yescrypt_rn(unsigned long count,
                     const uint8_t *rbytes, size_t nrbytes,
                     uint8_t *output, size_t o_size)
 {
-  /* Use one of recommended parameter sets as the 'low default'. */
-  yescrypt_params_t params = { .flags = YESCRYPT_DEFAULTS,
-                               .N = 4096, .r = 32, .p = 1
-                             };
+  /* Valid cost parameters are from 1 to 11.  The default is 5.
+     These are used to set yescrypt's 'N' and 'r' parameters as
+     follows:
+     N (block count) is specified in units of r (block size).
+     The author of yescrypt recommends in the documentation to use
+     r=8 for total sizes of 2 MiB and less, and r=32 above that.
+     This has to do with the typical per-core last-level cache sizes
+     of current CPUs.  */
```

Comment:
This comment have missed true explanation of parameters. `count` specifies memory usage of yescrypt. This memory usage is set up by setting `N` and `r` parameters. (In below code) that fact that `11 -> 262144` does not tell too much. Truly `11` is `262144 * 4096` i.e. `1073741824` i.e. 1 GiB of memory. It's better to say `11 -> 1 GiB` there (correcting other values too).